gemrb/plugins/SDLAudio/SDLAudio.cpp

```diff
@@ -73,7 +73,28 @@ bool SDLAudio::Init(void)
 	return true;
 }
 
-void SDLAudio::music_callback(void *udata, unsigned short *stream, int len) {
+void SDLAudio::SetAudioStreamVolume(uint8_t *stream, int len, int volume)
+{
+	uint8_t *mixData = new uint8_t[len];
+	memcpy(mixData, stream, len * sizeof(uint8_t));
+	memset(stream, 0, len); // mix audio data against silence
+	SDL_MixAudio(static_cast<uint8_t*>(stream), static_cast<const uint8_t*>(mixData), len, volume);
+	delete[] mixData;
```

Comment:
this all seems useless to me. am I misunderstanding?

looks like...
1. create a new buffer
2. copy existing stream into said buffer
3. set contents of current buffer to 0
4. mix new buffer into old buffer

can you really not do `SDL_MixAudio(stream, stream, len, volume);`

at the very lease those `static_cast`s are useless

Reply:
`SDL_MixAudio` is probably the only way of setting volume here that I know of
And to control the balance is a right way we have to mix against silence (mixing silence against existing stream results in the same stream, since silence * volume is well.. silence.
Changing `stream` pointer results in crash, so creation of empty mixData and doing some std::swap is not an option either.
I'm honestly not sure if it can be done any differently (except from modifying stream yourself in some way?).

static_casts are useless leftovers indeed tho. Removing them..

Reply:
sounds good, thanks